aws/update_costcenter_tag.py

```diff
@@ -46,130 +46,130 @@ def _mip_valid_tags():
     return json_data.json()
 
 
-def _resource_lookup(res_id):
+def _resource_tags(resource):
     """
-    Describe resource details of the given EC2 instance
-
-    TODO: support non-reserved instances; once we hit our monthly reservation,
-        the schema returned by ec2_client may change
-    TODO: support other types of resources (so far Cost Explorer has
-        only reported EC2 instances missing tags)
+    Get all tags associated with the given ARN or EC2 ID
     """
 
     results = []
-    data = ec2_client.describe_instances(InstanceIds=[res_id, ])
 
-    if data:
+    if resource.startswith("arn:"):
+        data = tag_client.get_resources(ResourceARNList=[resource, ])
+        logging.debug(data)
+        for tag_map in data["ResourceTagMappingList"]:
+            results.append(tag_map)
+    elif resource.startswith("i-"):
+        data = ec2_client.describe_instances(InstanceIds=[resource, ])
```

Comment:
EC2s can be identified by an ARN as well, https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-policy-structure.html#EC2_ARN_Format

Maybe that will allow you to simplify this new code?

Reply:
Yeah, I thought about removing the EC2 ID lookup and just using the ARN of the instance, but since one of the use cases for this script is updating tags on provisioned service catalog products on behalf of a user, and the web portal doesn't provide the ARNs of EC2 instances, we are likely to get requests from users that only include the instance ID of an EC2 instance in scipool. So I figured it was best to support both instead of relying on one of us manually building the EC2 ARN to pass to the script.

Also, I took a stab at programmatically building the EC2 ARN from the instance id, but I couldn't find a way to get the current account ID, and the next best way to get the ARN of an EC2 instance using boto is from `describe_instances`, which also returns the tags.

Reply:
does this help? https://stackoverflow.com/a/36709462

Reply:
yeah, that did, thanks